### ISSUE TITLE
[Octave] Error message for methods reports wrong number of arguments

### ISSUE THREAD

**Issue by alatina** [swig/swig#1210]
If one provides the wrong number of arguments to a class' method, the error message: e.g.
```
error: function MYCLASS_save_as_dst_file requires at least 3 arguments
```
is wrong:  the reported arguments is systematically +1 w.r.t. the real number of arguments required (in this example, it was 2). 

I believe this might be due to some "this->" pointer kind of thing, but yet it can be misleading for the final user.

**Comment by wsfulton** [comment]
Yes, this is the case for most languages which provide a proxy class wrapper on top of a flattened api where the first parameter is the 'this' pointer so while the message might be confusing, it is technically correct. Suggestions for better clarification welcome.
